src/NUnitFramework/framework/Constraints/ConstraintExpression.cs

```diff
@@ -66,6 +65,7 @@ public ConstraintExpression()
         /// <param name="builder">The builder.</param>
         public ConstraintExpression(ConstraintBuilder builder)
         {
+            Guard.ArgumentNotNull(builder, nameof(builder));
             this.builder = builder;
```

Comment:
All the code in this class assumes that `builder` is not null. Now that the field is readonly and this guard is in place, that is finally a safe assumption.